tests/unit/test_api.py

```diff
@@ -0,0 +1,29 @@
+
+import pytest
+import earthaccess
+from earthaccess import api,download
+from unittest.mock import Mock
+
+def test_download(monkeypatch):
+
+    earthaccess.login()
+
+    results = earthaccess.search_data(
+            short_name='ATL06',  
+            bounding_box=(-10, 20, 10, 50),  
+            temporal=("1999-02", "2019-03"), 
+            count=10
+    )
+
+    def mock_get(*args, **kwargs):
+        raise Exception('Download failed')
+
+    mock_store = Mock()
+    monkeypatch.setattr(earthaccess, '__store__', mock_store)
+    monkeypatch.setattr(mock_store, 'get', mock_get)
+
+    with pytest.raises(Exception, match="Download failed"):
+        earthaccess.download(results, "/home/download-folder")
+
+
```

Comment:
Please also add a 2nd test like so (I haven't tested, so you may need to adjust it).

We want to make sure that overriding the default pqdm kwargs works. We should probably try a few more things, but this should at least suffice for now:

```python
def test_download_deferred_failure(monkeypatch):
    earthaccess.login()

    results = earthaccess.search_data(
            short_name='ATL06',  
            bounding_box=(-10, 20, 10, 50),  
            temporal=("1999-02", "2019-03"), 
            count=10,
    )

    def mock_get(*args, **kwargs):
        raise Exception('Download failed')

    mock_store = Mock()
    monkeypatch.setattr(earthaccess, '__store__', mock_store)
    monkeypatch.setattr(mock_store, 'get', mock_get)

    results = earthaccess.download(results, "/home/download-folder",
            {"exception_behavior": "deferred"})

    assert len(results) == 10 and all(isinstance(e, Exception), results)
```